### PR TITLE
Support ANAME and fixup CNAME support

### PR DESCRIPTION
**Pull request by bluejekyll**
I'll file a separate PR for ANAME support in the resolver.

fyi: #513 

Also, it looks like CNAME was most likely broken before, where the server not even aliasing names properly before. This fixes that, and adds a bunch of tests for CNAME.

### CODE REVIEW

**Comment by bluejekyll** [comment]
FYI @moonshiner

File: crates/proto/src/rr/record_type.rs [hunk 1]
```diff
@@ -181,6 +184,8 @@ impl From<u16> for RecordType {
         match value {
             1 => RecordType::A,
             28 => RecordType::AAAA,
+            // FIXME: wrong value here
+            2005 => RecordType::ANAME,
```
Comment (pieterlexis) [inline crates/proto/src/rr/record_type.rs]: Random comment from a DNS-nerd: Please use a number in the "private" range (65280-65534) for types not yet [allocated by IANA](https://www.iana.org/assignments/dns-parameters/dns-parameters.xhtml#dns-parameters-4).

Comment (moonshiner) [reply crates/proto/src/rr/record_type.rs]: I second this.

Comment (bluejekyll) [reply crates/proto/src/rr/record_type.rs]: Will do. I was hoping there was an official value I hadn’t discovered yet.

Comment (moonshiner) [reply crates/proto/src/rr/record_type.rs]: I opened https://github.com/each/draft-aname/issues/48 to work on the RR Type application.

Comment (bluejekyll) [reply crates/proto/src/rr/record_type.rs]: Added a tracking issue: #723

**Comment by bluejekyll** [comment]
DNSSEC support for ANAME will come separately: #724

**Pull request merged**